ir-server/src/tls.rs

```diff
@@ -17,7 +18,42 @@ use crate::config::Config;
 use crate::utils;
 use crate::RegistryResult;
 
-fn tls_server_config() -> RegistryResult<Arc<ServerConfig>>
+enum TLSConfig
+{
+    TLS(Arc<ServerConfig>),
+    RaTLS(RealmMeasurements),
+}
+
+impl TLSConfig
+{
+    pub fn get_rustls_config(&self) -> RegistryResult<Arc<ServerConfig>>
+    {
+        match self {
+            Self::TLS(config) => Ok(config.clone()),
+            Self::RaTLS(reference_measurements) => {
+                let client_token_verifier = Arc::new(ChainVerifier::new(vec![
+                    // Arc::new(VeraisonTokenVerifer::new(
+                    //     &Config::readu().veraison_url,
+                    //     &Config::readu().veraison_pubkey,
+                    // )),
+                    Arc::new(RealmVerifier::init(reference_measurements.clone())),
+                ]));
+
+                let rustls_config = ServerConfig::builder()
+                    .with_client_cert_verifier(Arc::new(RaTlsCertVeryfier::from_token_verifier(
+                        client_token_verifier,
+                    )))
+                    .with_single_cert(
+                        utils::load_certificates_from_pem(&Config::readu().cert)?,
```

Comment:
Please cache those two files in the TLSConfig.

Reply:
Done.